base/io.jl

```diff
@@ -259,6 +259,9 @@ function peek(s::IO)
     end
 end
 
+peek(io, ::Type{UInt8}) = peek(io)
+peek(io, ::Type{Char}) = Char(peek(io))
```

Comment:
This is not generally correct: you need to look at up to four bytes to peek at a UTF-8 character.